app/components/square.js

```diff
@@ -0,0 +1,43 @@
+import Component from '@glimmer/component';
+import { assert } from '@ember/debug';
+
+export default class SquareComponent extends Component {
+  /**
+   * component arguments
+   *
+   * required
+   * @property row {Number} - the y coordinate of this square in the game board
+   * @property column {Number} - the x coordinate of this square in the game board
+   * @property ships {Array} - a [row] x [column] 2d array
+   * @property shots {Array} - a [row] x [column] 2d array
+   *
+   * optional
+   * @property shouldShowShip {Boolean}(default = false)
+   */
+
+  constructor() {
+    super(...arguments);
+    const { ships, shots, row, column } = this.args;
+    assert('row {Number} is required', Number.isInteger(row));
+    assert('column {Number} is required', Number.isInteger(column));
+    assert('ships {Array} is required', Array.isArray(ships));
```

Comment:
Unless we're going to pass an empty array for ships, I'd make it optional, since we're sometimes only going to show the shots for one of the boards.